oozie/src/main/java/org/apache/falcon/oozie/OozieBundleBuilder.java

```diff
@@ -106,11 +108,24 @@ public Properties build(Cluster cluster, Path buildPath) throws FalconException
         properties.setProperty(AbstractWorkflowEngine.NAME_NODE, ClusterHelper.getStorageUrl(cluster));
 
         //Add libpath
-        String libPath = getLibPath(buildPath);
-        if (libPath != null) {
-            properties.put(OozieClient.LIBPATH, getStoragePath(libPath));
+        String path = getLibPath(buildPath);
+        if (StringUtils.isNotBlank(path)) {
+            String storageLibPaths = null;
+            String[] libPaths = path.split(WF_LIB_SEPARATOR);
+            for (String libPath : libPaths) {
```

Comment:
In the for loop storageLibPaths isNotBlank() is getting multiple times to build a String for storageLibPaths. Shall not be this avoided appropriately.